x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -77,18 +95,22 @@ public void writeTo(StreamOutput out) throws IOException {
             @SuppressWarnings("unchecked")
             final String[] indices = ((List<String>) params[0]).toArray(String[]::new);
             final String analyticsCollectionName = (String) params[1];
-            return new Engine(engineName, indices, analyticsCollectionName);
+            final long updatedAt = (Long) params[2];
```

Comment:
This fails the test here:
```
Caused by: java.lang.NullPointerException: Cannot invoke \"java.lang.Long.longValue()\" because \"params[2]\" is null
23:40:27   1> 	at org.elasticsearch.xpack.entsearch.engine.Engine.lambda$static$1(Engine.java:98)
23:40:27   1> 	at org.elasticsearch.xcontent@8.8.0-
```
We need to test if `params[2]` is null first or use a `Long`.

Reply:
Thanks for the catch!